### DISCUSSION TITLE
SMTP email configuration: problems getting it working

### DISCUSSION THREAD

**Discussion by wonka929** [frangoteam/FUXA#956] [Q&A]
Good evening everyone and thanks a lot do all developers for this fantastic piece of software :)
I have a questions: i'm having many troubles configuring most of the smtp providers for sending emails (alerts and reports).

With google, mere username and password are always rejected. Outlook the same.

Is it due do the 2 factor authentication method that has been introduced all over the world?
What is the User/ApiKey field used for?
Is it the token i have to generate from (example) the google developer API?
Why should i need a password too if i have an API Key?

I'm unable to configure it....

**Comment by wonka929** [comment]
Here I am.
Sorry, i've found how to do it.

Basically, with the introduction by many platforms of the 2FA, the login on apps is managed by the general redirect to the login page of gmail and with that login they create a token which is then associated to the app.

With FUXA this functionality is not still present, so this is the workaround: **you need to create an APP-dedicated password for the login.**
 Here follows the guide:
https://support.google.com/mail/answer/185833?hl=en

When you have a dedicated password, just use the email + dedicated password for login to smtp inside of FUXA and it works like a charm.

**Reply by unocelli** [reply]
Awesome, I was not aware of this method, Thanks a lot !!!

**Comment by robertcarlssos-dot** [comment]
This is quite common with Gmail and Outlook now. Regular username/password often won't work because of their newer security requirements (2FA, App Passwords, or OAuth). If you want an easier setup, you can use [SMTPmart](https://smtpmart.com/). It offers simple SMTP authentication, reliable delivery, and is much easier to configure for sending emails.

Which SMTP provider are you using?
